examples/fit_ODNP_data.py

```diff
@@ -0,0 +1,177 @@
+# }}}
+with figlist_var() as fl:
+    # {{{Plot Ep
+    fl.next("Integrated Enhancement")
+    fl.plot(
+        Ep["power", :flip_idx],
+        "o",
+        label="Progressive Saturation Data",
+        capsize=6,
+        alpha=0.5,
+    )
+    fl.plot(
+        Ep["power", flip_idx:],
+        "rx",
+        label="Returning Power Check",
+        capsize=6,
+        alpha=0.5,
+    )
+    # }}}
+    # {{{ plot R1p data
+    fl.next(r"$R_{1}(p)$")
+    fl.plot(R1p, "o", label="Experimental Data")
+    # }}}
+    # {{{ fit kᵨ⁻¹ with two degrees of freedom (to a straight line) and then
+    # apply to fit R1p
+    T10_p = loadmat(
+        search_filename(
+            T100_info["filename"], exp_type=T100_info["data_dir"], unique=True
+        )
+    )["a"][0, :]
+    R10_p = 1 / (R1p.fromaxis("power").eval_poly(T10_p, "power"))
+    powers_fine = nddata(
+        r_[0 : R1p.getaxis("power") : 300j], "p"
+    )  # PR COMMENT: related to my comments below
+    krho_inv = Ep.get_prop("acq_params")["concentration"] / (R1p - R10_p)
+    krho_inv_coeff = krho_inv.polyfit("power", order=1)
+    krho_inv_fine = R1p.fromaxis("power").eval_poly(
+        krho_inv_coeff, "power"
+    )  # PR COMMENT: why is this called "fine" it's only evaluated at the powers present in the R1p data, which is not that many
+    M0, A, phalf, p = symbols("M0 A phalf power", real=True)
+    R1p_expression = (T10_p[0] + T10_p[1] * p) ** -1 + (
+        Ep.get_prop("acq_params")["concentration"]
+        / (krho_inv_coeff[0] + krho_inv_coeff[1] * p)
+    )
+    R1p_fit = lambdify(p, R1p_expression)
+    fl.plot(
+        R1p_fit(R1p.fromaxis("power")),
+        ls=":",  # PR COMMENT: again, why are you plotting this as :, since it's only evaluated at the powers of interest -- maybe you want to construct a new axis?  Why not just plot as a solid line, anyways?
+        color="k",
+        label="Fit",
+        alpha=0.5,
+    )
+    plt.ylabel(r"$R_{1} / s^{-1}$")
+    plt.xlabel("Power / W")
+    # }}}
+    # {{{ Fit E(p)
+    fl.next("Integrated Enhancement")
+    sp = p / (p + phalf)
+    Ep_fit = lmfitdata(Ep["power", :flip_idx])
+    # Symbolic expression for Ep that is used in the symbolic function for the fitting of E(p)
+    Ep_fit.functional_form = M0 - ((M0 * A * sp) / R1p_expression)
+    # generate a guess for the A parameter of the fit based on the normalized
+    # enhancement weighted by the relaxation rate. The bounds for the fit are
+    # then set to center around this value.
+    # PR COMMENT: edit the following as appropriate.
+    # Since
+    # E(pₘₐₓ) = 1 - A s(pₘₐₓ)/R₁(pₘₐₓ)
+    # and the max s(p) is about 1,
+    # 1-E(pₘₐₓ)R₁(pₘₐₓ) ≈ A
+    # because R₁(0) is typically larger and also easier to access than
+    # R₁(pₘₐₓ), we just use it instead.
+    A_guess = (
+        1
+        - (
+            R1p["power", 0].real.item()
+            * (
+                Ep["power", flip_idx].real.item()
+            )  # PR COMMENT: you were dividing by 1?
```

Comment:
no, the Ep is not normalized! without this dividion A is wayyyyyyyy too high of a guess!

Reply:
Remember here we are saying 1 - Ep NOT M0-M0*Ep